Bottom: 5a58559f172b39950c6753738f1e441507e25ad7
Top:    27ef4045a45bcbd5c246a40e60503925e7c11cc9
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-02-15 22:33:27 +0000

polygon.c: For some reason, the range computation in frac_circle appears to be wrong


---

diff --git a/src/polygon.c b/src/polygon.c
index 75e1b9e..40b15c3 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -404,8 +404,8 @@ frac_circle (PLINE * c, Coord X, Coord Y, Vector v, int fraction)
   e1 = (v[0] - X) * POLY_CIRC_RADIUS_ADJ;
   e2 = (v[1] - Y) * POLY_CIRC_RADIUS_ADJ;
 
-  /* NB: the caller adds the last vertex, hence the -1 */
-  range = POLY_CIRC_SEGS / fraction - 1;
+  /* XXX */ /* NB: the caller adds the last vertex, hence the -1 */
+  range = POLY_CIRC_SEGS / fraction;
   for (i = 0; i < range; i++)
     {
       /* rotate the vector */
